Lib/fontTools/ttLib/tables/TupleVariation.py

```diff
@@ -446,6 +444,118 @@ def getTupleSize_(flags, axisCount):
+		if coordWidth == 2:
+			self.coordinates = [
+				(d[0] * scalar, d[1] * scalar) if d is not None else None
+				for d in self.coordinates
+			]
+		elif coordWidth == 1:
+			self.coordinates = [
+				d * scalar if d is not None else None
+				for d in self.coordinates
+			]
+
+	def roundDeltas(self):
+		coordWidth = self.getCoordWidth()
+		if coordWidth == 2:
+			self.coordinates = [
+				(otRound(d[0]), otRound(d[1])) if d is not None else None
+				for d in self.coordinates
+			]
+		elif coordWidth == 1:
+			self.coordinates = [
+				otRound(d) if d is not None else None for d in self.coordinates
+			]
+
+	def calcInferredDeltas(self, origCoords, endPts):
+		from fontTools.varLib.iup import iup_delta
+
+		if self.getCoordWidth() == 1:
+			raise TypeError(
+				"Only 'gvar' TupleVariation can have inferred deltas"
+			)
+		if None in self.coordinates:
+			if len(self.coordinates) != len(origCoords):
+				raise ValueError(
+					"Expected len(origCoords) == %d; found %d"
+					% (len(self.coordinates), len(origCoords))
+				)
+			self.coordinates = iup_delta(self.coordinates, origCoords, endPts)
+
+	def optimize(self, origCoords, endPts, tolerance=0.5, isComposite=False):
+		from fontTools.varLib.iup import iup_delta_optimize
+
+		if None in self.coordinates:
+			return  # already optimized
+
+		deltaOpt = iup_delta_optimize(
+		    self.coordinates, origCoords, endPts, tolerance=tolerance
+		)
+		if None in deltaOpt:
+			if isComposite and all(d is None for d in deltaOpt):
+				# Fix for macOS composites
+				# https://github.com/fonttools/fonttools/issues/1381
+				deltaOpt = [(0, 0)] + [None] * (len(deltaOpt) - 1)
+			# Use "optimized" version only if smaller...
+			varOpt = TupleVariation(self.axes, deltaOpt)
+
+			# Shouldn't matter that this is different from fvar...?
+			axisTags = sorted(self.axes.keys())
+			tupleData, auxData, _ = self.compile(axisTags, [], None)
+			unoptimizedLength = len(tupleData) + len(auxData)
+			tupleData, auxData, _ = varOpt.compile(axisTags, [], None)
+			optimizedLength = len(tupleData) + len(auxData)
+
+			if optimizedLength < unoptimizedLength:
+				self.coordinates = varOpt.coordinates
+
+	def __iadd__(self, other):
+		if not isinstance(other, TupleVariation):
+			return NotImplemented
+		deltas1 = self.coordinates
+		length = len(deltas1)
+		deltas2 = other.coordinates
+		if len(deltas2) != length:
+			raise ValueError(
+				"cannot sum TupleVariation deltas with different lengths"
+			)
+		if self.getCoordWidth() == 2:
+			for i, d2 in zip(range(length), deltas2):
+				d1 = deltas1[i]
+				try:
+					deltas1[i] = (d1[0] + d2[0], d1[1] + d2[1])
+				except TypeError:
+					raise ValueError(
+						"cannot sum gvar deltas with inferred points"
+					)
+		else:
+			for i, d2 in zip(range(length), deltas2):
+				d1 = deltas1[i]
+				if d1 is not None and d2 is not None:
+					deltas1[i] = d1 + d2
+				elif d1 is None and d2 is not None:
+					deltas1[i] = d2
+				# elif d2 is None do nothing
```

Comment:
Is it possible for d1 to be None?

Reply:
yes, either could be None. `d1 is None` is already handled when `d2 is not None`; when both d1 and d2 are None, nothing happens.